html/inc/delete_account.inc

```diff
@@ -0,0 +1,187 @@
+<?php
+// This file is part of BOINC.
+// https://boinc.berkeley.edu
+// Copyright (C) 2018 University of California
+//
+// BOINC is free software; you can redistribute it and/or modify it
+// under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation,
+// either version 3 of the License, or (at your option) any later version.
+//
+// BOINC is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+// See the GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with BOINC.  If not, see <https://www.gnu.org/licenses/>.
+
+require_once("../inc/common_defs.inc");
+require_once("../inc/util.inc");
+require_once("../inc/user.inc");
+require_once("../inc/host.inc");
+require_once("../inc/friend.inc");
+require_once("../inc/boinc_db.inc");
+require_once("../inc/submit_util.inc");
+require_once("../project/project.inc");
+
+// Constants for different methods of deleting accounts
+// These correspond to the value used in the config.xml 
+// field of <enable_delete_account/>
+define("DELETE_ACCOUNT_METHOD_OBFUSCATE", 1);
+define("DELETE_ACCOUNT_METHOD_WIPE", 2);
+define("DELETE_ACCOUNT_METHOD_PROJECT_DEFINED", 3);
+
+$config = get_config();
+if ( !parse_bool($config, "enable_delete_account") ) {
+    error_page(
```

Comment:
@drshawnkwang how does Drupal handle calls to `error_page()`? Is this problematic since it's part of an `inc` file (API) but sounds like a front-end feature?

Reply:
Oliver is correct, this is not ideal from the Drupal standpoint. Drupal will call the `error_page()` function and display an error page, that is devoid of anything but the error message: think black text on white background.

Ideally, the config.xml check should happen elsewhere, either before the functions here are called. Or at the very beginning of each function, and then it returns false.

I don't know if that would require a major re-write.

Reply:
I've re-organized the code so calls to error_page only reside in view components.